src/wallet/load.cpp

```diff
@@ -117,3 +119,39 @@ void UnloadWallets()
         UnloadWallet(std::move(wallet));
     }
 }
+
+bool AddWalletSetting(interfaces::Chain& chain, const std::string& wallet_name)
+{
+    util::SettingsValue setting_value = chain.getRwSetting("wallet");
+    bool unset = setting_value.isNull();
+    if (!setting_value.isArray()) setting_value.setArray();
+    for (const util::SettingsValue& value : setting_value.getValues()) {
+        if (value.isStr() && value.get_str() == wallet_name) return true;
+    }
+    if (unset) {
+        // If dynamic wallet setting is being added for the first time, and the
+        // default SoftSetArg wallet is loaded, add the default wallet to the
+        // list so it still gets loaded next startup. This can be removed after
+        // https://github.com/bitcoin/bitcoin/pull/15454, when there is no more
+        // default SoftSetArg wallet.
```

Comment:
I don't think we can remove this behavior in #15454 as we still are loading the default wallet if it exists. So we still need this behavior there.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15937#discussion_r443866400

> I don't think we can remove this behavior in #15454 as we still are loading the default wallet if it exists. So we still need this behavior there.

Thanks, rethinking this and looking at #15454, there's no need to have a temporary workaround here at all if WalletInit::Construct initializes the wallet setting, so the PR now does that and drops this change.